featuretools/computational_backends/feature_set_calculator.py

```diff
@@ -822,6 +822,21 @@ def last_n(df):
                     to_merge = base_frame.groupby(groupby_col).agg(to_agg)
 
                 else:
+                    # TODO: Remove when https://github.com/pandas-dev/pandas/issues/57317 is fixed
+                    cols_to_fix = []
+                    for col in base_frame.columns:
+                        dtype = base_frame[col].dtype
+                        if (
+                            isinstance(dtype, pd.CategoricalDtype)
+                            and str(dtype.categories.dtype) == "int64"
+                        ):
+                            cols_to_fix.append(col)
+
+                    if cols_to_fix:
+                        base_frame[cols_to_fix] = base_frame[cols_to_fix].astype(
+                            "int64",
+                        )
```

Comment:
I ended up adding a fix for the bug where the groupby/aggregation is actually happening, because the `NumUnique` primitive itself just returns the pandas `pd.Series.nunique` function, and the bug is occurring at the actual aggregation step. This feels very removed from the actual bug, and I'm not fully in love with it as a workaround, so I'm open to other options. I'm a bit worried that it may be overly-broad; maybe there are numeric-category columns that we wouldn't want to turn into int64? But I didn't want to over engineer and add an explicit check for the nunique function.

Reply:
Probably worth waiting to merge this MR till one of the pandas folks responds to the bug I opened up in case they have a better workaround than what I've found

Reply:
Oh actually just got a response. It seems like just returning the string `"nunique"` in the primitive fixes this. Will investigate more on monday, but @thehomebrewnerd any idea why we weren't able to do that before? I see we do that for spark dataframes

Reply:
I'm not sure off the top of my head. I think it's possible that we could have done this before, but the paradigm is for the `get_function` method of a primitive to return a function, so that's probably why we did it that way. There are probably a few cases where both a string and a function could work and this must be one of those cases.

Reply:
The only resulting [failure](https://github.com/alteryx/featuretools/actions/runs/7850051684/job/21424702596?pr=2657) from making that change is in the docstring for the primitive, which must be the only place we attempt to calculate the value by instantiating and then using the primitive as a function. I'm leaning towards using 5dbc28d10e5ad9ca831e0d2cb14cf54a84f1ab24 as the fix here, as it is a much more minimal change than converting every numeric categorical column when we're calculating the feature matrix. 

@thehomebrewnerd any reason you can think of that we wouldn't be okay updating the docstring for now until the bug is fixed?

Reply:
Decided to add a param to allow users to use `pd.Series.nunique` if need be: 6d83654f431d49500645f0385f6d6b1c97636b99. If the pandas devs have plans to fix this issue, we'll eventually remove it. If the new behavior will eventually become the behvaior even for `'nunique'`, we'll need to implement a fix in `feature_set_calculator.py` to either convert integer categorical data or call `s.groupby(…).nunique(…)` directly when need be (as that method can keep the `int64` because it is aware that `nunique` would want the eventual type to be int (whereas `.agg` has no knowledge of the operations it's calling)